scripts/src/internal/services.ts

```diff
@@ -154,14 +156,14 @@ export async function paymentComplete(payment: CompletedPayment, logger: LoggerI
 				return;
 			} else {
 				order.value = asset.asOrderValue();
-				asset.ownerId = order.sender!.id;
+				asset.ownerId = order.user.id;
 				await asset.save();  // XXX should be in a transaction with order.save
 			}
 		}
-	} else if (order.isExternalOrder()) {
-		// XXX for p2p don't put the JWT in the recipient order's value
-		// XXX for p2p create a completed order for the recipient too
-		order.value = await getPaymentJWT(order, payment.app_id, order.type === "spend" ? order.sender!.id : order.recipient!.id);
+	} else if (order.isP2P()) {
+		order.value = await getPaymentJWT(order, payment.app_id, order.sender.id);
+	} else if (order.isNormal()) {
+		order.value = await getPaymentJWT(order, payment.app_id, order.user.id);
```

Comment:
Aren't these two if statements doing the same thing?

Reply:
Not sure this is the correct behaviour

Reply:
almost the same, but one uses `order.sender.id` while the other uses `order.user.id`.
when using `order.user` you don't care about whether he's the sender or receiver.